rules/DowngradePhp81/Rector/LNumber/DowngradeOctalNumberRector.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DowngradePhp81\Rector\LNumber;
+
+use PhpParser\Node;
+use PhpParser\Node\Scalar\LNumber;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://php.watch/versions/8.1/explicit-octal-notation
+ *
+ * @see \Rector\Tests\DowngradePhp81\Rector\LNumber\DowngradeOctalNumberRector\DowngradeOctalNumberRectorTest
+ */
+final class DowngradeOctalNumberRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Downgrades octal numbers to decimal ones', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return 0o777;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run()
+    {
+        return 0777;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [LNumber::class];
+    }
+
+    /**
+     * @param LNumber $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $numberKind = $node->getAttribute(AttributeKey::KIND);
+        if ($numberKind !== LNumber::KIND_OCT) {
+            return null;
+        }
+
+        $rawValue = $node->getAttribute(AttributeKey::RAW_VALUE);
+        if (! str_starts_with($rawValue, '0o')) {
+            return null;
+        }
+
+        $clearValue = str_replace('0o', '0', $rawValue);
```

Comment:
```suggestion
        $clearValue = '0' . ltrim((string) $rawValue, '0o')
```

see https://3v4l.org/VSSvg